.github/actions/npm-rc-release/action.yml

```diff
@@ -0,0 +1,37 @@
+name: 'Npm RC release'
+description: 'Publish code changes to npmjs.org'
+inputs:
+  git_email:
+    description: 'Email of git commit containing version bump'
+    required: true
+  git_username:
+    description: 'Commit author name'
+    required: true
+  npm_token:
+    description: 'Npm token for publishing. Login to npmjs.org to generate this.'
+    required: true
+runs:
+  using: 'composite'
+  steps:
+    - name: 'Configure git CLI'
+      shell: bash
+      run: |
+        git config --global user.email "${{ inputs.git_email }}"
+        git config --global user.name "${{ inputs.git_username }}"
+    - name: 'Bump version'
+      shell: bash
+      run: npm version prerelease --preid=rc --no-git-tag-version
+    - name: 'Commit changes'
+      shell: bash
+      run: |
+        git add .
+        newVersion=$(jq -r '.version' package.json)
+        git commit -m "[skip ci] Publish RC release v$newVersion"
```

Comment:
I think this will commit changes to whichever branch is targeted by the action. Possibly OK unless it'd be irritating to deal with when trying to port the inference stuff over to the public repo because we'd have these release commits to deal with.

I'm actually not sure if we should be committing or pushing at all, would the workflow work if we just run `npm version prerelease` and then `npm publish`?

Reply:
Okay, after speaking with @austin-denoble and @ssmith-pc offline, we have come to this conclusion, which we are implementing here: 
- if `main` is the latest release, it does not make sense to merge RC version commits into `main`;
- if `main` is the latest _development_ branch, it _does_ make sense to merge RC version commits there

Since our repo falls under the 1st bullet point, we will _not_ be pushing RC version commits to Github.